### ISSUE TITLE
axe-core Error: Cannot read property 'filter' of undefined

### ISSUE THREAD

**Issue by amiralitaheri** [dequelabs/axe-core#2483]
Running Lighthouse accessibility on [this url](https://amiralitaheri.ir/blog/2020/08/%DA%86%DA%AF%D9%88%D9%86%D9%87-%D8%AF%D8%B1-%D8%A8%D8%A7%D8%B2%DB%8C%D9%87%D8%A7%DB%8C-gameecom-%D8%AA%D9%82%D9%84%D8%A8-%DA%A9%D9%86%DB%8C%D9%85/), fails due axe-core error.

Actual: 
![Annotation 2020-08-26 021054](https://user-images.githubusercontent.com/45509862/91233793-2f458180-e747-11ea-9bb4-a82787645460.png)

Expectation: Stop failing!!

My runtime settings
![image](https://user-images.githubusercontent.com/45509862/91233950-93684580-e747-11ea-8b93-23a0c8e96b85.png)

**Comment by straker** [comment]
Can reproduce this. Seems the issue is with color contrast and the very long horizontal scroll of the Burp captured package output. We try to calculate the rect of the long string of text and calculate it using the full width and not the width of the container, causing our code to think it's much further to the right than it is.

**Comment by patrickhulce** [comment]
This error also occurs on https://www.nrk.no/sorlandet/fargesvake-ser-ikke-rod-t--onsker-seg-gul-1.15125048 if you need another repro case :)

ref https://github.com/GoogleChrome/lighthouse/issues/11384

**Comment by BogdanCerovac** [comment]
For https://www.nrk.no/sorlandet/fargesvake-ser-ikke-rod-t--onsker-seg-gul-1.15125048 

Raw Lighthouse CLI JSON report: https://pastebin.pl/view/a1fa102e
Raw aXe core CLI JSON report: https://pastebin.pl/view/339da0f9

Did not get any errors in the terminal (verbose mode).

**Comment by akinhwan** [comment]
I'm also seeing this error on my website when trying to run a lighthouse a11y test on desktop or mobile https://soceng.netlify.app/

**Comment by jferragut** [comment]
This is also present on our company website: [Alkemy](https://www.alkemydev.com)

**Reproduction:**
- Test Mobile in lighthouse
- Accessibility fails

**Screenshot:**
<img width="1680" alt="Screen Shot 2020-10-06 at 5 41 35 PM" src="https://user-images.githubusercontent.com/31458614/95263033-3312fc00-07fb-11eb-84f7-b4f890154779.png">

**Comment by straker** [comment]
I have tested both https://www.nrk.no/sorlandet/fargesvake-ser-ikke-rod-t--onsker-seg-gul-1.15125048 and https://www.alkemydev.com/ and found that the current axe version (v4.0.2) does not produce the error. Looking deeper it looks like both these issues were fixed in [v4.0.0](https://github.com/dequelabs/axe-core/releases/tag/v4.0.0), specifically with [a fix to handle text truncation](https://github.com/dequelabs/axe-core/pull/2302).

For Lighthouse this means they will need to update to v4.0.0 to resolve these issues. As for the original issue, it is not caused by text truncation and will be fixed in v4.1.0.

**Comment by padmavemulapati** [comment]
Verified in the latest develop code (v4.0.2)- and for the test snippet:
`<pre id="1" style="width: 400px; overflow: auto;">
		<span id="target" style="display: flex; with: 400px;">
		Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed et sollicitudin quam. Fusce mi odio, egestas pulvinar erat eget, vehicula tempus est. Proin vitae ullamcorper velit. Donec sagittis est justo, mattis iaculis arcu facilisis id. Proin pulvinar ornare arcu a fermentum. Quisque et dignissim nulla, sit amet consectetur ipsum. Donec in libero porttitor, dapibus neque imperdiet, aliquam est. Vivamus blandit volutpat fringilla. In mi magna, mollis sit amet imperdiet eu, rutrum ut tellus. Mauris vel condimentum nibh, quis ultricies nisi. Vivamus accumsan quam mauris, id iaculis quam fringilla ac. Curabitur pulvinar dolor ac magna vehicula, non auctor ligula dignissim. Nam ac nibh porttitor, malesuada tortor varius, feugiat turpis. Mauris dapibus, tellus ut viverra porta, ipsum turpis bibendum ligula, at tempor felis ante non libero. Donec dapibus, diam sit amet posuere commodo, magna orci hendrerit ipsum, eu egestas mauris nulla ut ipsum. Sed luctus, orci in fringilla finibus, odio leo porta dolor, eu dignissim risus eros eget erat.
		</span>
		</pre>`
I am still seeing the issue.
Confirmed with @straker , as per  discussion, moving back to "Inprogress"

**Comment by padmavemulapati** [comment]
Verified with the fix, from the develop branch code: (dated 11/11/2020 - with commit changes (https://github.com/dequelabs/axe-core/commit/3c2429b4032d29cdcfd3083e46e91854dccd733b)
Not seeing any failure on `color-contrast`
Verified for test snippet:
`<pre id="1" style="width: 400px; overflow: auto;">
		<span id="target" style="display: flex; with: 400px;">\n\n
		Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed et sollicitudin quam. Fusce mi odio, egestas pulvinar erat eget, vehicula tempus est. Proin vitae ullamcorper velit. Donec sagittis est justo, mattis iaculis arcu facilisis id. Proin pulvinar ornare arcu a fermentum. Quisque et dignissim nulla, sit amet consectetur ipsum. Donec in libero porttitor, dapibus neque imperdiet, aliquam est. Vivamus blandit volutpat fringilla. In mi magna, mollis sit amet imperdiet eu, rutrum ut tellus. Mauris vel condimentum nibh, quis ultricies nisi. Vivamus accumsan quam mauris, id iaculis quam fringilla ac. Curabitur pulvinar dolor ac magna vehicula, non auctor ligula dignissim. Nam ac nibh porttitor, malesuada tortor varius, feugiat turpis. Mauris dapibus, tellus ut viverra porta, ipsum turpis bibendum ligula, at tempor felis ante non libero. Donec dapibus, diam sit amet posuere commodo, magna orci hendrerit ipsum, eu egestas mauris nulla ut ipsum. Sed luctus, orci in fringilla finibus, odio leo porta dolor, eu dignissim risus eros eget erat.
		</span>
		</pre>`

![image](https://user-images.githubusercontent.com/47409579/98765796-e7a8ca80-2404-11eb-9919-15915d71410e.png)

**Comment by bnbabu55** [comment]
I am still getting this error using chrome/edge version 90, please help me get this taken care of.

![screenshot](https://user-images.githubusercontent.com/56356022/118486848-b6081780-b737-11eb-8dd9-9d6837ef3307.png)

**Comment by patrickhulce** [comment]
@bnbabu55 try with Chrome 91. If you're still seeing the issue then, it's a different root cause.

**Comment by davidstellini** [comment]
https://lighthouse-dot-webdotdevsite.appspot.com//lh/html?url=https%3A%2F%2Fallfront-marketing-site-qa.web.app%2F

I'm seeing this on web.dev, chrome 94 (canary) as well as Chrome 92.
Site is [All Front](https://allfront.io)
